### PR TITLE
[CU-2z0k7en] Removed excess storage read/writes from do_populate

### PR DESCRIPTION
**Pull request by PoisonPhang**
## Issue
[[CU-2z0k7en]](https://app.clickup.com/t/2z0k7en)

## Description
Removed unnecessary reads/writes from `do_populate` by only updating existing storage values as needed.

### CODE REVIEW

File: code/parachain/frame/crowdloan-rewards/src/lib.rs [hunk 1]
```diff
@@ -343,34 +343,51 @@ pub mod pallet {
...
-						remote_account,
-						Reward {
-							total: account_reward,
-							claimed: T::Balance::zero(),
-							vesting_period,
+
+			let mut total_rewards: T::Balance = TotalRewards::<T>::get();
+			let mut total_contributors: u32 = TotalContributors::<T>::get();
+
+			for (remote_account, total, vesting_period) in rewards.into_iter() {
```
Comment (vimukthi-git) [inline code/parachain/frame/crowdloan-rewards/src/lib.rs:363]:
```suggestion
			for (remote_account, account_total, vesting_period) in rewards.into_iter() {
```

**Review by vimukthi-git — APPROVED** [review]
LGTM :rocket:

File: code/parachain/frame/crowdloan-rewards/src/lib.rs (hunk 1 above)
Comment (PoisonPhang) [reply code/parachain/frame/crowdloan-rewards/src/lib.rs]: e73aa6a1b48bf1ec101fd8e2b3c2961588d17fb4

File: code/parachain/frame/crowdloan-rewards/src/lib.rs [hunk 2]
```diff
@@ -343,34 +343,54 @@ pub mod pallet {
...
-						remote_account,
-						Reward {
-							total: account_reward,
-							claimed: T::Balance::zero(),
-							vesting_period,
+
+			let mut total_rewards: T::Balance = TotalRewards::<T>::get();
+			let mut total_contributors: u32 = TotalContributors::<T>::get();
+
+			for (remote_account, account_total, vesting_period) in rewards.into_iter() {
```
Comment (benluelo) [inline code/parachain/frame/crowdloan-rewards/src/lib.rs:363]:
```suggestion
			for (remote_account, account_total, vesting_period) in rewards {
```

Comment (benluelo) [reply code/parachain/frame/crowdloan-rewards/src/lib.rs:363]: for loops call into_iter automatically

File: code/parachain/frame/crowdloan-rewards/src/lib.rs [hunk 3]
```diff
@@ -343,34 +343,54 @@ pub mod pallet {
...
+				Rewards::<T>::try_mutate_exists::<_, _, DispatchError, _>(
+					remote_account,
+					|reward| match reward {
+						Some(reward) => {
+							total_rewards = total_rewards.checked_sub(&reward.total).expect(
+								"TotalRewards is greater than or equal to reward.total; QED",
+							);
+							total_rewards = total_rewards.safe_add(&account_total)?;
```
Comment (benluelo) [inline code/parachain/frame/crowdloan-rewards/src/lib.rs:371]: chain these operations?

File: code/parachain/frame/crowdloan-rewards/src/lib.rs [hunk 4]
```diff
@@ -343,34 +343,54 @@ pub mod pallet {
...
-			)?;
+						None => {
+							total_contributors = total_contributors.safe_add(&1)?;
+							total_rewards = total_rewards.safe_add(&account_total)?;
+
+							*reward = Some(Reward {
+								total: account_total,
+								claimed: T::Balance::zero(),
+								vesting_period,
+							});
```
Comment (benluelo) [inline code/parachain/frame/crowdloan-rewards/src/lib.rs:386]: consider using https://doc.rust-lang.org/std/option/enum.Option.html#method.replace

File: code/parachain/frame/crowdloan-rewards/src/lib.rs [hunk 5]
```diff
@@ -343,34 +343,54 @@ pub mod pallet {
...
+			let mut total_rewards: T::Balance = TotalRewards::<T>::get();
+			let mut total_contributors: u32 = TotalContributors::<T>::get();
+
+			for (remote_account, account_total, vesting_period) in rewards.into_iter() {
+				Rewards::<T>::try_mutate_exists::<_, _, DispatchError, _>(
+					remote_account,
+					|reward| match reward {
+						Some(reward) => {
+							total_rewards = total_rewards.checked_sub(&reward.total).expect(
+								"TotalRewards is greater than or equal to reward.total; QED",
```
Comment (benluelo) [inline code/parachain/frame/crowdloan-rewards/src/lib.rs:369]: is there an assertion somewhere for this? or is this just an invariant i'm not aware of

Comment (PoisonPhang) [reply code/parachain/frame/crowdloan-rewards/src/lib.rs]: f9d8a8b6556ba7731ef70de608f3b9d2a5428d05

File: code/parachain/frame/crowdloan-rewards/src/lib.rs (hunk 4 above)
Comment (PoisonPhang) [reply code/parachain/frame/crowdloan-rewards/src/lib.rs]: f9d8a8b6556ba7731ef70de608f3b9d2a5428d05

File: code/parachain/frame/crowdloan-rewards/src/lib.rs (hunk 3 above)
Comment (PoisonPhang) [reply code/parachain/frame/crowdloan-rewards/src/lib.rs]: f9d8a8b6556ba7731ef70de608f3b9d2a5428d05

File: code/parachain/frame/crowdloan-rewards/src/lib.rs (hunk 2 above)
Comment (PoisonPhang) [reply code/parachain/frame/crowdloan-rewards/src/lib.rs]: f9d8a8b6556ba7731ef70de608f3b9d2a5428d05

**Review by benluelo — APPROVED** [review]
:rocket:

**Pull request merged**